### PR TITLE
[ARC-525] Reporting run results

### PR DESCRIPTION
**Pull request by SamTheisens**
branch: feature/ARC-525-report-results

### CODE REVIEW

**Comment by SamTheisens** [comment]
test

File: tests/stages/test_discovery.py [hunk 1]
```diff
@@ -28,7 +27,3 @@ def test_should_find_invalidated_dependencies(self):
                                                           [History(0, "revision", {'tests/projects/job/file.py',
                                                                                    'tests/some_file.txt'})])
         assert 1 == len(invalidated)
-
-
-if __name__ == '__main__':
-    unittest.main()
```
Comment (SamTheisens) [inline tests/stages/test_discovery.py:34]: Clean up python.unittest related stuff.

File: mpyl-test-runner.py [hunk 2]
```diff
@@ -17,15 +19,20 @@ class StepParam:
...
-    if not step_output.success:
-        raise Failure(description=step_output.message)
-    return step_output
+    dagster_logger = get_dagster_logger()
+    executor = Steps(dagster_logger, run_properties)
+    step_result = executor.execute(stage, proj)
+    # message = f'Executed {stage.name} for {proj.name}'
+    # dagster_logger.info(message)
+    # output = MplOutput(success=True, message=message)
+    # step_result = StepResult(stage=stage, project=proj, output=output)
```
Comment (SamTheisens) [inline mpyl-test-runner.py:32]: remove comment, or add flag to skip execution

File: Jenkinsfile [hunk 3]
```diff
@@ -12,6 +12,7 @@ pipeline {
         stage('Build') {
            environment {
                 DOCKER_REGISTRY = credentials('91751de6-20aa-4b12-8459-6e16094a233a')
+                GITHUB_TOKEN = credentials('github-pat-mpyl-vandebronjenkins')
```
Comment (SamTheisens) [inline Jenkinsfile:15]: Discuss governance over these secrets.

File: mpyl-test-runner.py (hunk 2 above)
Comment (SamTheisens) [reply mpyl-test-runner.py]: Solved in [cd77e12](https://github.com/Vandebron/mpyl/pull/28/commits/cd77e12967b68e8e2a1116cdc69e0594e141d6b3)

File: tests/stages/test_discovery.py (hunk 1 above)
Comment (SamTheisens) [reply tests/stages/test_discovery.py:34]: Solved in [2a71ca9](https://github.com/Vandebron/mpyl/pull/28/commits/2a71ca95e92221c6b575984f25ccd6df60a27218)

**Review by koeves — APPROVED** [review]
LGTM

**Review by ewaldschrader — APPROVED** [review]

**Comment by SamTheisens** [comment]
Stage BUILD
2023-02-17 16:01:19.712239 - job - Stage.BUILD - success: True 
2023-02-17 16:01:19.723691 - service - Stage.BUILD - success: True 
Stage DEPLOY
2023-02-17 16:01:20.704669 - job - Stage.DEPLOY - success: True 
2023-02-17 16:01:20.704669 - service - Stage.DEPLOY - success: True

**Pull request merged**